addons/hr_holidays/models/hr_leave_type.py

```diff
@@ -111,72 +102,20 @@ def _search_valid(self, operator, value):
-        result = dict((id, dict(max_leaves=0, leaves_taken=0, remaining_leaves=0, virtual_remaining_leaves=0)) for id in self.ids)
-
-        requests = self.env['hr.leave'].search([
-            ('employee_id', '=', employee_id),
-            ('state', 'in', ['confirm', 'validate1', 'validate']),
-            ('holiday_status_id', 'in', self.ids)
-        ])
-
-        allocations = self.env['hr.leave.allocation'].search([
-            ('employee_id', '=', employee_id),
-            ('state', 'in', ['confirm', 'validate1', 'validate']),
-            ('holiday_status_id', 'in', self.ids)
-        ])
-
-        for request in requests:
-            status_dict = result[request.holiday_status_id.id]
-            status_dict['virtual_remaining_leaves'] -= request.number_of_days_temp
-            if request.state == 'validate':
-                status_dict['leaves_taken'] += request.number_of_days_temp
-                status_dict['remaining_leaves'] -= request.number_of_days_temp
-
-        for allocation in allocations:
-            status_dict = result[allocation.holiday_status_id.id]
-            if allocation.state == 'validate':
-                # note: add only validated allocation even for the virtual
-                # count; otherwise pending then refused allocation allow
-                # the employee to create more leaves than possible
-                status_dict['virtual_remaining_leaves'] += allocation.number_of_days_temp
-                status_dict['max_leaves'] += allocation.number_of_days_temp
-                status_dict['remaining_leaves'] += allocation.number_of_days_temp
-
-        return result
-
-    @api.multi
-    def _compute_leaves(self):
-        data_days = {}
-        if 'employee_id' in self._context:
-            employee_id = self._context['employee_id']
-        else:
-            employee_id = self.env['hr.employee'].search([('user_id', '=', self.env.user.id)], limit=1).id
-
-        if employee_id:
-            data_days = self.get_days(employee_id)
-
-        for holiday_status in self:
-            result = data_days.get(holiday_status.id, {})
-            holiday_status.max_leaves = result.get('max_leaves', 0)
-            holiday_status.leaves_taken = result.get('leaves_taken', 0)
-            holiday_status.remaining_leaves = result.get('remaining_leaves', 0)
-            holiday_status.virtual_remaining_leaves = result.get('virtual_remaining_leaves', 0)
-
     @api.multi
     def name_get(self):
-        if not self._context.get('employee_id'):
+        employee_id = self.env['hr.employee'].browse(self._context.get('employee_id', 1))
+        if not employee_id:
             # leave counts is based on employee_id, would be inaccurate if not based on correct employee
             return super(HolidaysType, self).name_get()
         res = []
         for record in self:
+            employee_id = employee_id.with_context(holiday_status_id=record.ids)
```

Comment:
I would do the `browse` here, and call it `employee` and not `employee_id`.
```
employee = self.env['hr.employee'].browse(self._context['employee_id'])
```

To me, variables ending with `_id` should be actual ids (integers).
This does not apply when you assign a record to a many2one field of another record (e.g. user.partner_id = self.env['res.partner'].browse(1)`
That said, I am no longer sure of the Odoo standards and guidelines. I let @tde-banana-odoo confirm my thoughts.